core/roboconf-messaging-api/src/main/java/net/roboconf/messaging/internal/client/test/TestClientAgent.java

```diff
@@ -41,23 +44,16 @@
  */
 public class TestClientAgent implements IAgentClient {
 
-	public final List<Message> messagesForTheDm = new ArrayList<Message> ();
+	public final List<Message> messagesForTheDm = new ArrayList<> ();
 	public AtomicInteger messagesForAgentsCount = new AtomicInteger();
 	public AtomicBoolean connected = new AtomicBoolean( false );
 	public AtomicBoolean failMessageSending = new AtomicBoolean( false );
+	// TODO(?) : May be a good idea to make the previous parameters configurable via setConfiguration().
```

Comment:
+1 This may be quite an overkill!
On the other hand, the "test" factory may be a good example to illustrate how someone else could implement its own factory, re-configuration included.